backend/api/Controllers/MissionSchedulingController.cs

```diff
@@ -317,6 +327,15 @@ ISourceService sourceService
                 }
             }
 
+            IList<MissionStatus> missionStatuses = [MissionStatus.Ongoing, MissionStatus.Pending, MissionStatus.Paused];
+            var returnToHomeMission = await missionRunService.ReadReturnToHomeMissionRuns(robot.Id, missionStatuses);
+            if (returnToHomeMission != null)
+            {
+                await missionSchedulingService.StopCurrentMissionRun(robot.Id);
```

Comment:
I think we need to catch MissionRunNotFoundException here, since we do not lock this thread and the mission might therefore finish before we reach this. Or we need to lock this thread